### PR TITLE
Otel tracing

### PR DESCRIPTION
**Pull request by shanduur**

### CODE REVIEW

**Comment by zc-devs** [comment]
Is this configurable? How to enable / disable?
How does it look from admin perspective? Is this produces some enriched logs? Could you provide an example?
Maybe link some docs?

**Comment by shanduur** [comment]
Hi! @zc-devs this is still WIP, and only first draft is being made. I will write a full documentation, provide screenshots and usage examples later. Thanks for interest though!

**Comment by shanduur** [comment]
Looks like the build scripts are broken - cannot build UI as vite is missing?

```
make in_docker build-server
reuse existing docker image
run in docker:
(cd web/; pnpm install --frozen-lockfile; pnpm build)
Lockfile is up to date, resolution step is skipped
Packages: +559
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 ERR_PNPM_ENOENT  ENOENT: no such file or directory, copyfile '/build/.pnpm-store/v3/files/b7/9d4ba102e1f43e088f77fbb33d524597ad6b00b3c5e5d96525a081d8cbc6d616107c4f0dc96c7cf98356c188c2379643cc4042c284edeb5a648df8b091d46b' -> '/build/web/node_modules/.pnpm/es-abstract@1.23.2/node_modules/es-abstract_tmp_12/2022/abs.js'
Progress: resolved 559, reused 468, downloaded 1, added 7
Downloading @iconify/json@2.2.196: 2.41 MB/70.86 MB

> woodpecker-ci@0.0.0 build /build/web
> vite build --base=/BASE_PATH

sh: vite: not found
 ELIFECYCLE  Command failed.
make: *** [Makefile:196: build-ui] Error 1
make: *** [in_docker] Error 2
```

**Comment by anbraten** [comment]
Haven't used the `make in_docker` by myself. Might be required to run `pnpm vite`

File: shared/tracing/tracing.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+		}
+
+		wg.Wait()
+	}
+}
+
+func registerTraceExporter(res *resource.Resource, exporter sdktrace.SpanExporter) (func(context.Context) error, error) {
+	options := []sdktrace.TracerProviderOption{
+		sdktrace.WithBatcher(exporter),
+		sdktrace.WithSampler(sdktrace.AlwaysSample()),
```
Comment (lafriks) [inline shared/tracing/tracing.go:77]: This should probably not be used (at least should remember to remove it when PR is done it it's used currently for debugging purposes)

**Pull request closed without merging**